_posts/2019-09-30-introducing-arrow-flight.md

```diff
@@ -0,0 +1,260 @@
+be transferred to local hosts before being deserialized.
+
+The work we have done since the beginning of Apache Arrow holds exciting
+promise for accelerating data transport in a number of ways. The [Arrow
+columnar format][2] has key features that can help us:
+
+* It is an "on-the-wire" representation of tabular data that does not require
+  deserialization on receipt
+* Its natural mode is that of "streaming batches", larger datasets are
+  transported a batch of rows at a time (called "record batches" in Arrow
+  parlance). In this post we will talk about "data streams", these are
+  sequences of Arrow record batches using the project's binary protocol
+* The format is language-independent and now has library support in 11
+  languages and counting. For
+
+Implementations of standard protocols like ODBC generally implement their own
+custom on-wire binary protocols that must be marshalled to and from each
+library's public interface. The performance of ODBC or JDBC libraries varies
+greatly from case to case.
+
+Our design goal for Flight is to create a new protocol for data services that
+uses the Arrow columnar format as both the over-the-wire data representation as
+well as the public API presented to developers. In doing so, we reduce or
+remove the serialization costs associated with data transport and increase the
+overall efficiency of distributed data systems. Additionally, two systems that
+are already using Apache Arrow for other purposes can communicate data to each
+other with extreme efficiency.
+
+## Flight Basics
+
+The Arrow Flight libraries provide a development framework for implementing a
+service that can send and receive data streams. A Flight server supports
+several basic kinds of requests:
+
+* **ListFlights**: return a list of available data streams
+* **GetSchema**: return the schema for a data stream
+* **GetFlightInfo**: return a "query plan" for a dataset of interest, possibly
+  requiring consuming multiple data streams. This request can accept custom
+  serialized commands containing, for example, your specific application
+  parameters.
+* **DoGet**: send a data stream to a client
+* **DoPut**: receive a data stream from a client
+* **DoAction**: a perform an implementation-specific action and
+  return any results, i.e. a generalized function call
+* **ListActions**: return a list of available action types
+
+A simple Flight setup might consist of a single server to which clients connect
+and make DoGet requests.
+
+<div align="center">
+<img src="{{ site.baseurl }}/img/20191014_flight_simple.png"
+     alt="Flight Simple Architecture"
+     width="50%" class="img-responsive">
+</div>
+
+## Optimizing Data Throughput over gRPC
+
+While using a general-purpose messaging library like gRPC has numerous specific
+benefits beyond the obvious ones (taking advantage of all the engineering that
+Google has done on the problem).
+
+The best-supported way to use gRPC is to define services using an extended
```

Comment:
Why "extended version"?

Reply:
It's because keywords like `service` and `rpc` used in Flight.proto are not built in to Protocol Buffers -- you need to use the grpc_cpp_plugin when invoking the protoc compiler to generate the gRPC service stubs

Reply:
I amended the language to clarify